lib/ts/fetch.ts

```diff
@@ -382,34 +397,38 @@ export async function onUnauthorisedResponse(
                     return { result: "SESSION_EXPIRED" };
                 }
                 if (
-                    postLockID.status !== preRequestIdToken.status ||
-                    (postLockID.status === "EXISTS" &&
-                        preRequestIdToken.status === "EXISTS" &&
-                        postLockID.token !== preRequestIdToken.token)
+                    postLockLSS.status !== preRequestLSS.status ||
+                    (postLockLSS.status === "EXISTS" &&
+                        preRequestLSS.status === "EXISTS" &&
+                        postLockLSS.lastRefreshAttempt !== preRequestLSS.lastRefreshAttempt)
                 ) {
                     logDebugMessage(
                         "onUnauthorisedResponse: Retrying early because pre and post id refresh tokens don't match"
                     );
                     // means that some other process has already called this API and succeeded. so we need to call it again
                     return { result: "RETRY" };
                 }
-                let headers: any = {};
-                if (preRequestIdToken.status === "EXISTS") {
-                    const antiCsrfToken = await AntiCsrfToken.getToken(preRequestIdToken.token);
+
+                await saveRefreshAttempt();
```

Comment:
Im not sure if we should call this here. I think we should call this after refresh is successful cause:
- User is using cookie based auth in safari.
- Safari clears all frontend cookies - including last refresh time (refresh token in httpOnly cookies still exists)
- This causes a refresh attempt to happen which saves the last refresh time to now.
- The refresh API is called but it fails (500 error for example) resulting in none of the cookies of the front token to be set.
- The if `getLocalSessionState` is called again, it will return `NOT_EXISTS` even if the refresh token exists in cookies.